app/src/main/java/io/left/meshim/controllers/RightMeshController.java

```diff
@@ -442,4 +468,77 @@ public void run() {
             },UNDELIVERED_PACKAGE_TIMEOUT);
         }
     }
-}
\ No newline at end of file
+
+    /**
+     * this functions converts a byte array into a file
+     * @param fileName name of the file with extension
+     * @param fileByte file bytes to convert
+     * @return a path to the newly written file
+     */
+    public String writeFileExternalStorage(String fileName, byte[] fileByte) {
+        String state = Environment.getExternalStorageState();
+        //external storage availability check
+        if (!Environment.MEDIA_MOUNTED.equals(state)) {
+            return null;
+        }
+        File file = new File(Environment.getExternalStoragePublicDirectory(
+                Environment.DIRECTORY_DOWNLOADS), fileName);
+        FileOutputStream outputStream = null;
+        try {
+            file.createNewFile();
+            //second argument of FileOutputStream constructor indicates whether to append or create new file if one exists
+            outputStream = new FileOutputStream(file, false);
+            outputStream.write(fileByte);
+            outputStream.flush();
+            outputStream.close();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return file.getPath();
+    }
+
+    /**
+     * this function turns any file into byte arrays
+     * @param file the file to turn byte arrays into
+     * @return  byte arrays
+     * @throws IOException
+     */
+    public static byte[] getBytesFromFile(File file) throws IOException {
+        InputStream fileInputStream = new FileInputStream(file);
+
+        // Get the size of the file
+        long length = file.length();
+
+        // Before converting to an int type, check
+        // to ensure that file is not larger than Integer.MAX_VALUE.
+        if (length > Integer.MAX_VALUE) {
+            // File is too large
+        }
+
+        // Create the byte array to hold the data
+        byte[] bytes = new byte[(int)length];
+        // Read in the bytes
+        int offset = 0;
+        int numRead = 0;
+        while (offset < bytes.length
+                && (numRead=fileInputStream.read(bytes, offset, bytes.length-offset)) >= 0) {
+            offset += numRead;
+        }
+        // Ensure all the bytes have been read in
+        if (offset < bytes.length) {
+            throw new IOException("Could not completely read file "+file.getName());
+        }
+        // Close the input stream and return bytes
+        fileInputStream.close();
+        return bytes;
+    }
+
+    public static String getFileExtension(String fileName) {
+        String extension = null;
```

Comment:
You should probably document how this method works, especially that it can return null. Would an empty string be more appropriate maybe? Or maybe the whole filename? (E.g. `Makefile` is a make file, even without a dot at the end.)